packages/nx-sonarqube/src/executors/scan/utils/utils.ts

```diff
@@ -128,10 +153,19 @@ export async function scanner(
     };
   }
 
-  await sonarQubeScanner.async({
+  scannerOptions = combineOptions(
+    new ExtraMarshaller(options.extra),
+    new EnvMarshaller(),
+    scannerOptions
+  );
+  const success = await sonarQubeScanner.async({
     serverUrl: options.hostUrl,
     options: scannerOptions,
   });
+  return {
+    success: success,
+    scannerOptions: scannerOptions,
```

Comment:
executors should only return `{success: success}`

Reply:
This is the function that that does the scanning. The executor function is unchanged  and returns {success: success}. The reasoning behind this change is so that I can see if the scannerOptions respects the extraOptions and environment variables set by the user. 

Another way to check if the extra options and environment variables are respected without changing the function signature is to move it a different function. 
 `function getScannerOptions(options:ScanExecutorSchema,sources:string, lcovPaths:string, branch: string):  { [option: string]: string } {
  let scannerOptions: { [option: string]: string } = {
    'sonar.exclusions': options.exclusions,
    'sonar.javascript.lcov.reportPaths': lcovPaths,
    'sonar.language': 'ts',
    'sonar.login': process.env.SONAR_LOGIN,
    'sonar.organization': options.organization,
    'sonar.password': process.env.SONAR_PASSWORD,
    'sonar.projectKey': options.projectKey,
    'sonar.projectName': options.projectName,
    'sonar.projectVersion': options.projectVersion,
    'sonar.qualitygate.timeout': options.qualityGateTimeout,
    'sonar.qualitygate.wait': String(options.qualityGate),
    'sonar.scm.provider': 'git',
    'sonar.sources': sources,
    'sonar.sourceEncoding': 'UTF-8',
    'sonar.tests': sources,
    'sonar.test.inclusions': options.testInclusions,
    'sonar.typescript.tsconfigPath': 'tsconfig.base.json',
    'sonar.verbose': String(options.verbose),
  };
  if (options.branches) {
    scannerOptions["sonar.branch.name"] = branch
  }
  scannerOptions = combineOptions(
    new ExtraMarshaller(options.extra),
    new EnvMarshaller(),
    scannerOptions
  )
  return scannerOptions
}`

I prefer to move it to a different function as it would ease testing but I did not want to change the code too much.